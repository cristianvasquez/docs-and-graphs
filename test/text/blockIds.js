import { expect } from 'expect'

import toMatchSnapshot from 'expect-mocha-snapshot'
import {
  extractBlockIds, removeBlockIds,
} from '../../src/text/blockIds.js'

const tweets = [
  '## A header with some ^id',
  '^id1 and ^id2']


expect.extend({ toMatchSnapshot })

describe('extractBlockIds', async function () {
  for (const current of tweets) {
    it(current, async function () {
      const tags = extractBlockIds(current)
      expect(tags).toMatchSnapshot(this)
    })
  }
})

describe('removeBlockIds', async function () {
  for (const current of tweets) {
    it(current, async function () {

      const txt = removeBlockIds(current)
      expect(txt).toMatchSnapshot(this)
    })
  }
})
