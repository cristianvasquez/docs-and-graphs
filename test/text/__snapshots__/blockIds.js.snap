// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`extractBlockIds ## A header with some ^id 1`] = `
Array [
  "^id",
]
`;

exports[`extractBlockIds ^id1 and ^id2 1`] = `
Array [
  "^id1",
  "^id2",
]
`;

exports[`removeBlockIds ## A header with some ^id 1`] = `"## A header with some "`;

exports[`removeBlockIds ^id1 and ^id2 1`] = `" and "`;
